common/include/pcl/common/io.h

```diff
@@ -319,10 +319,10 @@ namespace pcl
                   pcl::PCLPointCloud2 &cloud_out);
 
   /** \brief Check if two given point types are the same or not. */
-  template <typename Point1T, typename Point2T> inline bool
+  template <typename Point1T, typename Point2T> inline constexpr bool
```

Comment:
should this function live in `common/include/pcl/type_traits.h` perhaps?

Reply:
We can simplify this a lot more (and deprecate it) since this is just a function call instead of `std::is_same_v<A, B>`. Thoughts: @mvieth ?

Reply:
I am not necessarily against deprecating `isSamePointType`, but I don't really see a problem with people using `pcl::isSamePointType<A, B>()` instead of `std::is_same<A, B>::value` (or `is_same_v` starting with C++17).
How would you simplify it?

Reply:
Alright. If we don't deprecate it, should we move it to `type_traits.h` or some other file than `io.h`?

Reply:
Apart from this, LGTM